### PR TITLE
cmpjs api v1.1

### PR DESCRIPTION
**Pull request by braizhas**
- Support for CMP JS API v1.1 - #32
- Toolbox testing mode
- Some refactoring

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -1,4 +1,108 @@
 {
   "cmpVersion": 1,
-  "cmpId": 1
+  "cmpId": 1,
+  "maxCookieLifespanDays" : 300,
```
Comment (SupahNickie) [inline metadata.json]: This should be 390 days

File: src/lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
+//jshint esversion: 6
```
Comment (SupahNickie) [inline src/lib/utils.js]: Will you please add tests that cover the same functions moved to the new `utils` file?

Comment (braizhas) [reply src/lib/utils.js]: sure, working on that.

File: src/lib/cmp.js [hunk 3]
```diff
@@ -38,7 +38,13 @@ export default class Cmp {
 				const publisherConsents = (config.storePublisherData && store.getPublisherConsentsObject()) || { lastUpdated: Date.now() }; // if publisher consent is not enabled mark - cookie as valid
 				const shouldBePromted = checkReprompt(config.repromptOptions, vendorConsents, publisherConsents);
 
-				if (config.gdprAppliesGlobally) {
+				if (config.testingMode) {
```
Comment (SupahNickie) [inline src/lib/cmp.js]: We want three cases for a testing mode:
- Set to always show, so the CMP renders every time you load the page no matter what
- Set to never show, so the CMP has to be triggered with Javascript in the console ( `window.__cmp('showConsentTool'` )
- Normal

**Pull request merged**